core/sqf/src/tm/tmddlrequests.cpp

```diff
@@ -34,57 +34,68 @@ using namespace std;
 * Signature: ([B)V
 */
 
-JNIEXPORT jint JNICALL Java_org_apache_hadoop_hbase_client_transactional_RMInterface_createTableReq
-  (JNIEnv *pp_env, jobject pv_object, jbyteArray pv_tableDescriptor, jobjectArray pv_keys, jint pv_numSplits, jint pv_keyLength, jlong pv_transid, jbyteArray pv_tblname){
+JNIEXPORT jstring JNICALL Java_org_apache_hadoop_hbase_client_transactional_RMInterface_createTableReq
+  (JNIEnv *pp_env, jobject pv_object, jbyteArray pv_tableDescriptor, jobjectArray pv_keys, jint pv_numSplits, jint pv_keyLength, jlong pv_transid, jbyteArray pv_tblname)
+{
 
-   char la_tbldesc[TM_MAX_DDLREQUEST_STRING];
-   char la_tblname[TM_MAX_DDLREQUEST_STRING];
-   char* str_key;
-   str_key = new char[pv_keyLength];
+   char *la_err_str = 0;
+   int la_err_len = 0;
    char** la_keys;
-   la_keys = new char *[TM_MAX_DDLREQUEST_STRING];
    int lv_error = FEOK;
 
-   int lv_tblname_len = pp_env->GetArrayLength(pv_tblname);
-   if(lv_tblname_len > TM_MAX_DDLREQUEST_STRING) {
-      cout << "Table name length is larger than max allowed" << endl;
+   int lv_tbldesc_length = pp_env->GetArrayLength(pv_tableDescriptor);
+   if(lv_tbldesc_length > TM_MAX_DDLREQUEST_STRING)
+   {
+     jstring lv_err_str = pp_env->NewStringUTF("Table Desc length is larger than max allowed");
```

Comment:
But there is a return at line 50  below. So, I would think it is leak when there is an error. Similar issue might exist else where in other functions too